addons/account/report/account_invoice_report.py

```diff
@@ -153,47 +123,33 @@ def _where(self):
                 AND NOT line.exclude_from_invoice_tab
         '''
 
-    @api.model
-    def _group_by(self):
-        return '''
-            GROUP BY
-                line.id,
-                line.move_id,
-                line.product_id,
-                line.account_id,
-                line.analytic_account_id,
-                line.journal_id,
-                line.company_id,
-                line.currency_id,
-                line.partner_id,
-                move.name,
-                move.state,
-                move.move_type,
-                move.amount_residual_signed,
-                move.amount_total_signed,
-                move.partner_id,
-                move.invoice_user_id,
-                move.fiscal_position_id,
-                move.payment_state,
-                move.invoice_date,
-                move.invoice_date_due,
-                move.invoice_payment_term_id,
-                move.invoice_partner_bank_id,
-                uom_template.id,
-                template.categ_id,
-                COALESCE(partner.country_id, commercial_partner.country_id)
-        '''
-
     def init(self):
         tools.drop_view_if_exists(self.env.cr, self._table)
         self.env.cr.execute('''
             CREATE OR REPLACE VIEW %s AS (
-                %s %s %s %s
+                %s %s %s
             )
         ''' % (
-            self._table, self._select(), self._from(), self._where(), self._group_by()
+            self._table, self._select(), self._from(), self._where()
         ))
 
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
+        @lru_cache(maxsize=32)  # cache to prevent a SQL query for each data point
+        def get_rate(currency_id):
+            return self.env['res.currency']._get_conversion_rate(
+                self.env['res.currency'].browse(currency_id),
+                self.env.company.currency_id,
+                self.env.company,
+                self._fields['invoice_date'].today()
+            )
+        result = super(AccountInvoiceReport, self).read_group(domain, fields, set(groupby) | set(['company_currency_id']), offset, limit, orderby, False)
+        for res in result:
+            if self.env.company.currency_id.id != res['company_currency_id'][0]:
+                for field in set(['price_average', 'price_subtotal']) & set(res):
```

Comment:
Same as on 13.0 branch; use {} for the set

Reply:
Indeed